File: python/0169. majorityElement.py
# naive
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        return collections.Counter(nums).most_common(1)[0][0]

# sort
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        return nums[len(nums)//2]

# Boyer-Moore majority voting
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        m, i = None, 0

        for x in nums:
            if i == 0:
                m = x
                i = 1
            elif m == x:
                i += 1
            else:
                i -= 1

        return m

File: rust/0169. majority_element.rs
// Boyer-Moore majority voting
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut m = 0;
        let mut i = 0;

        for x in nums {
            if i == 0 {
                m = x;
                i = 1;
            } else if m == x {
                i += 1
            } else {
                i -= 1
            }
        }

        m
    }
}
